central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -0,0 +1,244 @@
+package complianceReportgenerator
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+	checkResults "github.com/stackrox/rox/central/complianceoperator/v2/checkresults/datastore"
+	"github.com/stackrox/rox/central/graphql/resolvers"
+	"github.com/stackrox/rox/central/graphql/resolvers/loaders"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/branding"
+	"github.com/stackrox/rox/pkg/csv"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/grpc/authz/allow"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/notifier"
+	"github.com/stackrox/rox/pkg/notifiers"
+	"github.com/stackrox/rox/pkg/retry"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/templates"
+)
+
+var (
+	log = logging.LoggerForModule()
+
+	reportGenCtx = resolvers.SetAuthorizerOverride(loaders.WithLoaderContext(sac.WithAllAccess(context.Background())), allow.Anonymous())
+
+	csvHeader = []string{
+		"Control Reference",
+		"Check(CCR)",
+		"Profile(version)",
+		"Check Description",
+		"Cluster",
+		"Status",
+		"Remediation",
+	}
+)
+
+const (
+	maxNumberProfilesinSubject = 4
```

Comment:
Check comment above where we can summarize x number of profiles in subject to avoid the need for this